### PR TITLE
Add tests for ButtonMixin

### PR DESCRIPTION
**Pull request by margaree**
Resolves https://github.com/BrightspaceUI/core/issues/107

### CODE REVIEW

File: components/button/test/button-mixin.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+				});
+
+				it('should call button focus', async() => {
+					return new Promise(resolve => {
+						button.shadowRoot.querySelector('button').addEventListener('focus', resolve, {once: true});
+						button.focus();
+					});
+				});
+
+				it('should stop propegation if button is disabled', async() => {
```
Comment (svanherk) [inline components/button/test/button-mixin.html]: Minor: "propagation"

File: components/button/test/button-mixin.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+					});
+				});
+
+				it('should stop propagation if button is disabled', async() => {
+					const disabledButton = fixture('disabled');
+					const childButton = disabledButton.querySelector('test-button-elem');
+					await childButton.updateComplete;
+
+					return new Promise((resolve, reject) => {
+						disabledButton.addEventListener('click', () => {
```
Comment (dlockhart) [inline components/button/test/button-mixin.html]: You might actually be able to just add the event listener to the document/body.

File: components/button/test/button-mixin.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+							reject();
+						}, {once: true});
+						childButton.click();
+						setTimeout(resolve, 2000); // pass test if parent click eventlistener never called
```
Comment (dlockhart) [inline components/button/test/button-mixin.html]: In practice is 2s needed? Seems like the bubbling should happen pretty quickly.

Comment (margaree) [reply components/button/test/button-mixin.html]: No, it's pretty instantaneous. I'll decrease this.

**Comment by dlockhart** [comment]
Looks like everything is working! 🎉

**Pull request merged**